federation/cluster/common.sh

```diff
@@ -74,17 +74,16 @@ function cleanup-federation-api-objects {
   set +o errexit
 
   echo "Cleaning Federation control plane objects"
-  # Delete all resources with the federated-cluster label.
-  $host_kubectl delete pods,svc,rc,deployment,secret -lapp=federated-cluster
-
-  # Delete all PVs bound to PVCs in FEDERATION_NAMESPACE
-  pvs=$($host_kubectl get pvc --namespace=${FEDERATION_NAMESPACE} -o jsonpath='{.items[*].spec.volumeName}')
-  while $host_kubectl delete pv ${pvs} >/dev/null 2>&1; do
-    sleep 2
+  kube::log::status "Removing namespace \"${FEDERATION_NAMESPACE}\" from \"${FEDERATION_KUBE_CONTEXT}\""
+  # Try deleting until the namespace is completely gone.
+  while $host_kubectl --context="${FEDERATION_KUBE_CONTEXT}" delete namespace "${FEDERATION_NAMESPACE}" >/dev/null 2>&1; do
+    # It is usually slower to remove a namespace because it involves
+    # performing a cascading deletion of all the resources in the
+    # namespace. So we sleep a little longer than other resources
+    # before retrying
+    sleep 5
```

Comment:
Can we rather wait with a polled query on the ns; and bigger timeout which fails, if it could not delete the ns in that time.
Also I am curious as to why this was not done earlier. (probably some specific reason @madhusudancs )

Reply:
> Can we rather wait with a polled query on the ns; and bigger timeout which fails, if it could not delete the ns in that time.

@irfanurrehman What Shashi has now is actually the preferred method in kubernetes. 

> Also I am curious as to why this was not done earlier. (probably some specific reason @madhusudancs )

It's mostly for historic reasons. We wanted to be super careful about what we were deleting in the `federation-cluster` (now `federation-system`) namespace in the  pre-kubefed days. It just carried over to kubefed days.

Reply:
@madhusudan thanks for the clarifications.